1-select-tasks/4.txt

```diff
@@ -2,4 +2,14 @@ Create a query to return:
 
 | Customer Id | Total number of Orders | % of all orders |
  
-order data by % - higher percent at the top.
\ No newline at end of file
+order data by % - higher percent at the top.
+
+USE Northwind
+;
+
+SELECT CustomerID
+, COUNT(*) AS TotalNumberOfOrders
+, CAST ( COUNT(*) AS DECIMAL ) / SUM(COUNT(*)) OVER() * 100 AS [% of all orders]
```

Comment:
don't cast use literal of decimal type. See what's literal on wiki https://en.wikipedia.org/wiki/Literal_(computer_programming). They can be of different types in sql